### PR TITLE
Proposed implementation of issue #504

### PR DESCRIPTION
**Pull request by jeremiah-c-leary**
Configuration documentation has been updated for proposed changes to line above rule.

### CODE REVIEW

File: docs/configuring_previous_line_rules.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
++=====================+==================================================+
+| require_blank_line  | Requires a blank line on the line above.         |
++---------------------+--------------------------------------------------+
+| no_code             | Either a blank or comment on the line above.     |
++---------------------+--------------------------------------------------+
+| allow_comment       | Either a blank or comment on line above, with a  |
+|                     | blank line above the comment.                    |
++---------------------+--------------------------------------------------+
+| require_comment     | Requires a comment on the line above, with a     |
+|                     | blank line above the comment.                    |
```
Comment (staerz) [inline docs/configuring_previous_line_rules.rst:20]: Sorry, first time giving multi line suggestions on github - I think I missed the header and last line ...
```suggestion
| require_blank_line  | Requires a blank line on the line above.                   |
+---------------------+------------------------------------------------------------+
| no_code             | Either a blank line; or comment(s) on the line(s) above.   |
+---------------------+------------------------------------------------------------+
| allow_comment       | Either a blank line; or comment(s) on line(s) above and a  |
|                     | blank line above the comment(s).                           |
+---------------------+------------------------------------------------------------+
| require_comment     | Comment(s) required on the line(s) above and a             |
|                     | blank line above the comment(s).                           |
```

File: docs/configuring_previous_line_rules.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+   rule :
+     entity_003:
+        method : require_blank_line
+
+.. NOTE:: All examples below are using the rule **entity_004**.
+
+Example: require_blank_line
+###########################
+
+The following code would fail with this option:
```
Comment (staerz) [inline docs/configuring_previous_line_rules.rst:36]: I think you should always give the fail example first, then the pass.

This is in alignment with the general rule violations: you first always show the violation and then the fix.

Comment (staerz) [reply docs/configuring_previous_line_rules.rst:36]: hm ... ok, the difference here is that it's not a fix that you suggest, right? ...

Well, in that case, simply make the examples have pass and fail in the same order :wink:

**Comment by staerz** [comment]
Well, you came up with a very good name, I think `method` is very good!

I'd say the documentation is good like now (modulo my few suggestions modulo rearranging the table after I messed up the suggestion :rofl: )

File: docs/configuring_previous_line_rules.rst (hunk 2 above)
Comment (jeremiah-c-leary) [reply docs/configuring_previous_line_rules.rst:36]: I thought about making it a fix, but thought just illustrating what would trigger and what would not trigger the rule would be  clearer.

**Comment by jeremiah-c-leary** [comment]
> I'd say the documentation is good like now

Awesome...I will start working on the changes.

**Pull request merged**